packages/contracts/src/UnirepContract.ts

```diff
@@ -0,0 +1,134 @@
+        const Unirep = this.getUnirepArtifact(artifactsPath)
+        const addrMap = {}
+
+        for (const verifier of Object.keys(CircuitName)) {
+            const verifierName = this.createVerifierName(verifier)
+            const artifact = require(path.join(
+                artifactsPath,
+                'contracts',
+                'verifiers',
+                `${verifierName}.sol`,
+                `${verifierName}.json`
+            ))
+
+            console.log(`Deploying ${verifierName}`)
+            const factory = new ethers.ContractFactory(
+                artifact.abi,
+                artifact.bytecode,
+                deployer
+            )
+            const contract = await factory.deploy()
+            await contract.deployTransaction.wait()
+
+            addrMap[verifier] = contract.address
+        }
+
+        console.log('Deploying Unirep')
+        const f = new ethers.ContractFactory(
+            Unirep.abi,
+            Unirep.bytecode,
+            deployer
+        )
+        const c = await f.deploy(
+            {
+                globalStateTreeDepth: config.globalStateTreeDepth,
+                userStateTreeDepth: config.userStateTreeDepth,
+                epochTreeDepth: config.epochTreeDepth,
+            },
+            {
+                maxUsers: config.maxUsers,
+                maxAttesters: config.maxAttesters,
+            },
+            addrMap[CircuitName.verifyEpochKey],
+            addrMap[CircuitName.startTransition],
+            addrMap[CircuitName.processAttestations],
+            addrMap[CircuitName.userStateTransition],
+            addrMap[CircuitName.proveReputation],
+            addrMap[CircuitName.proveUserSignUp],
+            config.numEpochKeyNoncePerEpoch,
+            config.maxReputationBudget,
+            config.epochLength,
+            config.attestingFee
+        )
+
+        await c.deployTransaction.wait()
+
+        // Print out deployment info
+        console.log(
+            '-----------------------------------------------------------------'
+        )
+        console.log(
+            'Bytecode size of Unirep:',
+            Math.floor(Unirep.bytecode.length / 2),
+            'bytes'
+        )
+        const receipt = await c.provider.getTransactionReceipt(
+            c.deployTransaction.hash
+        )
+        console.log('Gas cost of deploying Unirep:', receipt.gasUsed.toString())
+        console.log(
+            '-----------------------------------------------------------------'
+        )
+
+        return c as Unirep
+    }
+
+    /**
+     * Get Unirep contract from a specific address
+     * @param address The address of the Unirep contract
+     * @param signerOrProvider A given signer or provider
+     * @param artifactsPath If artifacts path is given, it loads abi from the given artifacts path.
+     * Or it loads Unirep abi from exported package
+     * @returns The Unirep contract object
+     */
+    public static get = (
```

Comment:
I think we're falling into the trap of classes that don't really abstract here. Why would I learn this:

```
import UnirepContract from '@unirep/contracts`
import { UNIREP_ADDRESS } from './constants'

const signer = {create signer}
const contract = UnirepContract.get(address, signer)
```

When this does the same thing:

```
import { ethers } from 'ethers'
import { UnirepABI } from '@unirep/contracts'
import { UNIREP_ADDRESS } from './constants'

const signer = {create signer}
const unirepContract = new ethers.Contract(UNIREP_ADDRESS, UnirepABI, signer)
```

Even though we get some type support with typescript it's still sometimes hard to tell what a variable is. If I get an object of type `UnirepContract` it's hard to tell that it's just an ethers contract. The code in the second example is very familiar; I immediately know what I can and can't do with the `unirepContract` object. This is another case of letting developers use shared tooling (ethers) so they can learn the code more quickly.